### DISCUSSION TITLE
How to allow user to Connect Twitter / LinkedIn account (get access token permanently) after user is logged-in to the next-js app using next-auth's Google provider?

### DISCUSSION THREAD

**Discussion by aypen-git** [nextauthjs/next-auth#6217] [Help]
## Scenario :
1. User is logged in on my app using Google oauth (using next-auth Google provider)

2. Now I want to give user the option to connect their Twitter account permanently on Button click(i.e. store their Twitter access token somewhere in db).

3. Once the user connects and authorises my app using Twitter, my app can perform action on Twitter on behalf of the user.

4. Next time, user just has to login on my app using google auth, and he should be able to access and perform action (use Twitter API) without authorising again from twitter page

## Main Question:

Q1. Does next-auth support different providers on different page? If yes, then how to call signIn() for each of them individually?

Q2. Can I implement point 2 to point 4 in next-auth? If yes, then how? If No, then what's the alternative approach?

**Comment by aypen-git** [comment]
Would love to know the answers to the above question!

**Reply by NverKhachatryan** [reply]
Hey @Ayushpandey80 . Have you found an answer to this question?

**Reply by rexfordessilfie** [reply]
Hey. You might find this [discussion](https://github.com/nextauthjs/next-auth/discussions/1702) on a similar topic helpful.
